pkg/kubelet/network/dns/dns.go

```diff
@@ -241,19 +247,99 @@ func (c *Configurer) GetPodDNS(pod *v1.Pod) (*runtimeapi.DNSConfig, error) {
 			return nil, err
 		}
 	}
-	useClusterFirstPolicy := ((pod.Spec.DNSPolicy == v1.DNSClusterFirst && !kubecontainer.IsHostNetworkPod(pod)) || pod.Spec.DNSPolicy == v1.DNSClusterFirstWithHostNet)
-	if useClusterFirstPolicy && len(c.clusterDNS) == 0 {
+	return &runtimeapi.DNSConfig{
+		Servers:  hostDNS,
+		Searches: hostSearch,
+		Options:  hostOptions,
+	}, nil
+}
+
+func (c *Configurer) getPodDNSType(pod *v1.Pod) podDNSType {
```

Comment:
This should really return `(podDNSType, error)`.

Return the appropriate error if we end up with an incompatible combination1, then put the event generation and defaulting to ClusterFirst outside this function.

Reply:
Good point, moved the logic out and modified `GetPodDNS()` a bit, PTAL